### PR TITLE
Domain picker & focused launch: UI and data changes

### LINKED ISSUE #46862: Focused Launch: Domain step
If the site doesn't have a paid domain associated:
  - [x] Show the domain picker in the "active" group
  - [x] Step title is "Confirm your domain"
  - [x] Side commentary:
    - [x] On desktop, reads  "46.9% of globally registered domains are .com" and shows additional list items — PR #47281
    - [x] On mobile, starts with a bulb icon, reads "46.9% of registered domains are .com" and doesn't show any list items — PR #47281
  - [x] Show 3 domains (one is the default .wordpress.com)
  - [x] Show a "View all domains" link which changes the view to the domains picker details
  - [x] New "domain item" design - PR #47152  
  - [x] Show blue highlight for selected domain item - PR #47152 
  - [x] Pre-select the domain, by looking at the `@automattic/launch` data-store - PR #47191
  - [ ] If the `@automattic/launch` data-store doesn't contain a domain, fallback to items in the cart (if any) | #47233 
  - [ ] Ensure selected domain is always shown to the user (append it as the last paid domain if not in the initial list) #47540

Screenshot 2020-11-06 at 16 04 29

Screenshot 2020-10-28 at 15 13 42

Screenshot 2020-11-10 at 15 48 39

If the site already has a pre-purchased domain associated:
  - [x] show the domain in the "disabled" group
  - [x] Step title is 'Your domain'
  - [x] Show an info icon with a popover next to the title that reads "Changes to your purchased domain can be managed from your Domains page."
  - [x] Show only the purchased domain, which can't be selected by the user
  - [x] Side commentary:
    - [x] On desktop, reads  "Unique domains help build brand recognition and trust" — PR #47281
    - [x] On mobile, starts with a bulb icon, and reads "Unique domains help build brand trust" — PR #47281

Screenshot 2020-11-06 at 16 04 15

Screenshot 2020-11-10 at 15 48 09

### PR DESCRIPTION
**Pull request by alshakero**
This PR implements the necessary changes to focused launch and domain picker to have them work together without breaking any other usage of the domain picker.

#### Todo

- [x] Implement the data-store changes needed to get the current domain status of the site.
- [x] Implement the already purchased custom domain UI.
- [x] Only 3 domains are shown (one is the free domain)
- [x] Implement the dynamic steps count (finished steps don't count as steps). Eg if the user has a domain, step lables look like this: (1. Name your site, Your domain, 2. Pick a plan). 
- [x] Implement the [info tip icon](https://user-images.githubusercontent.com/1083581/97448093-8aa31400-1930-11eb-9c05-9b074619e4a1.png) when the domain is already purchased. 

#### Testing instructions - requires sandboxing 

1. In apps/editing-toolkit run `yarn dev --sync`.
2. Sandbox testingfocusedflow.wordpress.com.
3. Visit https://testingfocusedflow.wordpress.com/wp-admin/post-new.php.
4. In console, type `wp.data.dispatch('automattic/launch' ).openFocusedLaunch()`.

Fixes some of #46862

### CODE REVIEW

File: packages/launch/src/focused-launch/index.tsx [hunk 1]
```diff
@@ -67,28 +73,39 @@ const FocusedLaunch: React.FunctionComponent< Props > = ( { siteId } ) => {
...
+									{ __( 'Your domain', __i18n_text_domain__ ) }
+								</label>
+								<LockedPurchasedItem domainName={ sitePrimaryDomain?.domain || '' } />
+							</>
+						) : (
+							<DomainPicker
+								header={
+									<>
+										<label className="focused-launch__label">
+											{ stepNumber++ }. { __( 'Confirm your domain', __i18n_text_domain__ ) }
```
Comment (a8ci18n) [inline packages/launch/src/focused-launch/index.tsx:88]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **7** times:
`translate( '2. Confirm your domain' )` *ES Score: 9*

File: packages/launch/src/focused-launch/index.tsx [hunk 2]
```diff
@@ -112,7 +129,7 @@ const FocusedLaunch: React.FunctionComponent< Props > = ( { siteId } ) => {
 				<div className="focused-launch__data-input">
 					<div className="focused-launch__section">
 						<label className="focused-launch__label">
-							{ __( '3. Confirm your plan', __i18n_text_domain__ ) }
+							{ stepNumber++ }. { __( 'Confirm your plan', __i18n_text_domain__ ) }
```
Comment (a8ci18n) [inline packages/launch/src/focused-launch/index.tsx:132]: :warning: This change will be queued for retranslation. We'll display the English text until then.

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 3]
```diff
@@ -75,31 +80,39 @@ const Summary: React.FunctionComponent< Props > = ( { siteId } ) => {
...
+									{ __( 'Your domain', __i18n_text_domain__ ) }
+								</label>
+								<LockedPurchasedItem domainName={ sitePrimaryDomain?.domain || '' } />
+							</>
+						) : (
+							<DomainPicker
+								header={
+									<>
+										<label className="focused-launch-summary__label">
+											{ stepNumber++ }. { __( 'Confirm your domain', __i18n_text_domain__ ) }
```
Comment (a8ci18n) [inline packages/launch/src/focused-launch/summary/index.tsx:95]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **7** times:
`translate( '2. Confirm your domain' )` *ES Score: 9*

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 4]
```diff
@@ -123,7 +136,7 @@ const Summary: React.FunctionComponent< Props > = ( { siteId } ) => {
 				<div className="focused-launch-summary__data-input">
 					<div className="focused-launch-summary__section">
 						<label className="focused-launch-summary__label">
-							{ __( '3. Confirm your plan', __i18n_text_domain__ ) }
+							{ stepNumber++ }. { __( 'Confirm your plan', __i18n_text_domain__ ) }
```
Comment (a8ci18n) [inline packages/launch/src/focused-launch/summary/index.tsx:139]: :warning: This change will be queued for retranslation. We'll display the English text until then.

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 5]
```diff
@@ -75,31 +80,39 @@ const Summary: React.FunctionComponent< Props > = ( { siteId } ) => {
...
-							showSearchField={ false }
-							analyticsFlowId="focused-launch"
-							analyticsUiAlgo="focused_launch_domain_picker"
-							onDomainSearchBlur={ () => noop( 'TODO: on domain search blur' ) }
-							onSetDomainSearch={ () => noop( 'TODO: on set domain search' ) }
-							quantity={ 3 }
-						/>
-						<Link to={ Route.DomainDetails }>
-							{ __( 'View all domains', __i18n_text_domain__ ) }
-						</Link>
```
Comment (razvanpapadopol) [inline packages/launch/src/focused-launch/summary/index.tsx:102]: The link to `DomainDetails` was probably removed by mistake during merge.

**Comment by alshakero** [comment]
This is ready for review. I'll leave the last two items to a separate PR. This one grew slightly too big.

**Review by tjcafferkey — APPROVED** [review]
This works for me.

However we don't seem to be doing anything in the `catch` if the resolvers request fails? How will this impact the UI, do you think it's worth creating an issue for?

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@alshakero please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/942732). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

File: packages/domain-picker/src/domain-picker/locked-purchased-item.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+		<div className="domain-picker__suggestion-item locked">
+			<div className="domain-picker__suggestion-item-name">{ domainName }</div>
+			<div className="domain-picker__price">
+				<Icon icon={ check } size={ 18 } /> { __( 'Purchased', __i18n_text_domain__ ) }
+			</div>
+		</div>
+	);
+};
+
+export default LockedPurchasedItem;
```
Comment (razvanpapadopol) [inline packages/domain-picker/src/domain-picker/locked-purchased-item.tsx:19]: This component doesn't seem to have anything to do with `domain-picker` package. We need the exact same UI component also for `plans-grid`.
Screenshot 2020-11-05 at 14 19 11

Created issue https://github.com/Automattic/wp-calypso/issues/47147

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 7]
```diff
@@ -75,28 +88,48 @@ const Summary: React.FunctionComponent< Props > = ( { siteId } ) => {
...
+									{ __( 'Your domain', __i18n_text_domain__ ) }
+									<Tooltip
+										position="top center"
+										text={ __(
+											'Changes to your purchased domain can be managed from your Domains page.',
+											__i18n_text_domain__
+										) }
+									>
+										{ info }
+									</Tooltip>
```
Comment (razvanpapadopol) [inline packages/launch/src/focused-launch/summary/index.tsx:103]: There is already a reusable component for this. It needs to be extracted to `@automattic/components`.
https://github.com/Automattic/wp-calypso/issues/45449

**Pull request merged**